[Chakra upgrade v3] Make sure that the desktop nav menu lazyloads its content (@W-18989161@)

vmarta opened the pull request:
This PR makes sure to preserve the original behaviour, which is the initially-hidden content of the nav menu is lazyloaded. When the shoppers open the menu, that is when the app would fetch the menu content.

TODOs
- [x] open menu on click (not on hover)

## How to test-drive the PR

The issue was happening on desktop only.
1. Navigate to http://localhost:3000/ on desktop
2. Click on the New Arrivals category
3. A loading spinner would appear, as the subcategories are being fetched
4. Click on another top-level category, say, Womens
5. Expect to see another loading spinner too

vmarta committed:
Make sure that the desktop nav menu lazyloads its content

By not initially rendering the content and render it when users want to open the menu.

cc-prodsec commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/0923cb96-f92a-4a20-a9d6-ae2db1249d1d?refs=a118318603ebe935fac353845d7b06bfd74143a2&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/0923cb96-f92a-4a20-a9d6-ae2db1249d1d/license?refs=a118318603ebe935fac353845d7b06bfd74143a2&source=prChecksComment)

vmarta commented on packages/extension-chakra-storefront/src/components/list-menu/list-menu-popover.jsx:
```diff
@@ -39,18 +39,20 @@ const ListMenuPopover = ({contentComponent, item, name, itemsKey, maxColumns}) =
-                        </Popover.Body>
-                    </Popover.Content>
-                </Popover.Positioner>
+                {open && (
```
Before:

```jsx
<Popover.Root lazyMounted>
  <Popover.Trigger>...</Popover.Trigger>
  <Popover.Content>           <!-- Always rendered -->
    <ContentComponent />      <!-- Data always loaded -->
  </Popover.Content>
</Popover.Root>
```

After: 

```jsx
<Popover.Root lazyMounted>
  <Popover.Trigger>...</Popover.Trigger>
  {open && (                  <!-- Only rendered when open -->
    <Popover.Content>
      <ContentComponent />    <!-- Data only loaded when needed -->
    </Popover.Content>
  )}
</Popover.Root>
```

alexvuong replied:
I thought the lazyMounted is it would render when open, and dismount on closing. 🤔

vmarta replied:
@alexvuong I see what you mean 🤔 It looks like the official doc has a typo when it mentions `lazyMounted`: https://www.chakra-ui.com/docs/components/popover#lazy-mount 

It's actually `lazyMount` 😅 Let me try that.

vmarta committed:
Turns out that the Chakra prop on official doc has a typo

vmarta committed:
Now opening menu on click (no longer on hover)

We wanted to use the default behaviour of Chakra v3 popover, which is on click.

vmarta commented on packages/extension-chakra-storefront/src/components/list-menu/list-menu-popover.jsx:
```diff
@@ -14,45 +14,35 @@ import {ListMenuTrigger} from '../../components/list-menu/list-menu-trigger'
-        </Box>
+        <Popover.Root
+            positioning={{placement: 'bottom-start'}}
+            lazyMount
```
It used to be `lazyMounted`. Chakra's official doc has a typo. The correct spelling is `lazyMount`.

vmarta committed:
Clean up code

vmarta commented on packages/extension-chakra-storefront/src/components/list-menu/list-menu-popover.jsx:
```diff
@@ -14,45 +14,33 @@ import {ListMenuTrigger} from '../../components/list-menu/list-menu-trigger'
     const styles = recipe()
 
     return (
-        <Box onMouseLeave={onClose}>
```
We now removed the customization to open the menu on hover. Now preferring to rely on the default click behaviour of Chakra popover.
